xml/System.IO/Directory.xml

```diff
@@ -1039,12 +1080,63 @@ Directory::CreateDirectory("Public\\Html");
+
+ `searchPattern` cannot end in two periods ("..") or contain two periods ("..") followed by <xref:System.IO.Path.DirectorySeparatorChar> or <xref:System.IO.Path.AltDirectorySeparatorChar>, nor can it contain any invalid characters. You can query for invalid characters by using the <xref:System.IO.Path.GetInvalidPathChars%2A> method.
+
+ You can specify relative path information with the `path` parameter. Relative path information is interpreted as relative to the current working directory, which you can determine by using the <xref:System.IO.Directory.GetCurrentDirectory%2A> method.
+
+ The <xref:System.IO.Directory.EnumerateFiles%2A> and <xref:System.IO.Directory.GetFiles%2A> methods differ as follows: When you use <xref:System.IO.Directory.EnumerateFiles%2A>, you can start enumerating the collection of names before the whole collection is returned; when you use <xref:System.IO.Directory.GetFiles%2A>, you must wait for the whole array of names to be returned before you can access the array. Therefore, when you are working with many files and directories, <xref:System.IO.Directory.EnumerateFiles%2A> can be more efficient.
+
+ The returned collection is not cached; each call to the <xref:System.Collections.Generic.IEnumerable%601.GetEnumerator%2A> on the collection will start a new enumeration.
+  
+ ]]></format>
+        </remarks>
+        <exception cref="T:System.ArgumentException">
+          <paramref name="path" /> is a zero-length string, contains only white space, or contains invalid characters. You can query for invalid characters by using the <see cref="M:System.IO.Path.GetInvalidPathChars" /> method.  
+  
+ -or-  
+  
+ <paramref name="searchPattern" /> does not contain a valid pattern.</exception>
+        <exception cref="T:System.ArgumentNullException">
+          <paramref name="path" /> is <see langword="null" />.  
+  
+ -or-  
+  
+ <paramref name="searchPattern" /> is <see langword="null" />.</exception>
+        <exception cref="T:System.ArgumentOutOfRangeException">
+          <paramref name="searchOption" /> is not a valid <see cref="T:System.IO.SearchOption" /> value.</exception>
+        <exception cref="T:System.IO.DirectoryNotFoundException">
+          <paramref name="path" /> is invalid, such as referring to an unmapped drive.</exception>
+        <exception cref="T:System.IO.IOException">
+          <paramref name="path" /> is a file name.</exception>
+        <exception cref="T:System.IO.PathTooLongException">The specified path, file name, or combined exceed the system-defined maximum length.</exception>
+        <exception cref="T:System.Security.SecurityException">The caller does not have the required permission.</exception>
+        <exception cref="T:System.UnauthorizedAccessException">The caller does not have the required permission.</exception>
```

Comment:
both exceptions have the same text (I realize that some of these might have been here before as well). 

is there a way to clarify when you'd get one vs. the other?

Reply:
UnauthorizedAccessException is thrown in .NET Core for sure, when the user does not have permission to access a particular folder.

I don't recall where I saw SecurityException. I can't find it in .NET Framework, which is where I was suspecting it would come from.

I'll remove it.